src/tests/unittests/test_pgeometry.py

```diff
@@ -89,11 +89,26 @@ def test_polyintersect_degenerate_polygon(self):
         x1 = np.array([(0, 0)])
         x2 = np.array([(1, 0), (1.5, 1.5)])
 
-        try:
+        with self.assertRaises(ValueError):
             intersection_polygon = pgeometry.polyintersect(x1, x2)
-            self.assertEmptyPolygon(intersection_polygon)
-        except PolygonError as ex: # the assertRaises does not work due to the ctypes error message
-            self.assertIn('Invalid polygon or contour for operation', str(ex))
+
+    def test_multi_polygon_intersection(self):
+        delta = .42
+        shift = 1.7
+        x1 = np.array([(-1, -1), (1, -1), (1, 1), (-1, 1)]) + np.array([shift, 0])
+        x2 = np.array([(delta, 0), (5, 5), (-5, 5), (-delta, 0), (-5, -5), (5, -5), (delta, 0)])
+
+        with self.assertRaises(Exception):
```

Comment:
When you did change it to ValueError also here we can do assertRaises(ValueError):
```suggestion
        with self.assertRaises(ValueError):
```